diff --git a/ripl/mn.py b/ripl/mn.py
old mode 100644
new mode 100755
index 2af6558..ead6b2b
--- a/ripl/mn.py
+++ b/ripl/mn.py
@@ -8,8 +8,10 @@ To use this file to run a RipL-specific topology on Mininet.  Example:
 """
 
 from ripl.dctopo import FatTreeTopo #, VL2Topo, TreeTopo
+from mininet.topolib import FatTreeTopo1
 
-topos = { 'ft': FatTreeTopo}
+topos = { 'ft': FatTreeTopo,
+          'fattree': FatTreeTopo1 }
 #,
 #          'vl2': VL2Topo,
 #          'tree': TreeTopo }
