--- /usr/share/kernel-package/ruleset/misc/version_vars.mk	2010-03-25 18:14:41.000000000 -0700
+++ version_vars.mk	2010-03-03 06:46:59.000000000 -0800
@@ -138,11 +138,13 @@
 EXTRAV_ARG :=
 endif
 
-UTS_RELEASE_HEADER=$(call doit,if [ -f include/linux/utsrelease.h ]; then  \
+UTS_RELEASE_HEADER=$(call doit, if [ -f include/generated/utsrelease.h ]; then \
+	                       echo include/generated/utsrelease.h;            \
+			   else if [ -f include/linux/utsrelease.h ]; then  \
 	                       echo include/linux/utsrelease.h;            \
 	                   else                                            \
                                echo include/linux/version.h ;              \
-	                   fi)
+	                   fi fi)
 UTS_RELEASE_VERSION=$(call doit,if [ -f $(UTS_RELEASE_HEADER) ]; then                    \
                  grep 'define UTS_RELEASE' $(UTS_RELEASE_HEADER) |                       \
                  perl -nle  'm/^\s*\#define\s+UTS_RELEASE\s+("?)(\S+)\1/g && print $$2;';\
